flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/SpillableSubpartitionTest.java

```diff
@@ -193,7 +194,7 @@ public void notifyBuffersAvailable(long numBuffers) {
 	public void testConsumeSpilledPartition() throws Exception {
 		SpillableSubpartition partition = createSubpartition();
 
-		Buffer buffer = new Buffer(MemorySegmentFactory.allocateUnpooledSegment(4096), FreeingBufferRecycler.INSTANCE);
+		Buffer buffer = new NetworkBuffer(MemorySegmentFactory.allocateUnpooledSegment(4096), FreeingBufferRecycler.INSTANCE, true, 4096);
```

Comment:
`TestBufferFactory.createBuffer(4096);`

probably same could be applied to many some places?

Reply:
actually, that's what the included `[hotfix][tests] make SpillableSubpartitionTest use TestBufferFactory.createBuffer` does...

Reply:
oh - a classical race condition here - I fixed this one day before your review and put this onto the branch ;)